src/site/content/en/fast/efficiently-load-third-party-javascript/index.md

```diff
@@ -0,0 +1,174 @@
+
+An alternative approach is to load third-party content only when users scroll down to that section of the page.
+
+[Intersection Observer](https://developers.google.com/web/updates/2016/04/intersectionobserver) is a browser API that efficiently detects when an element enters or exits the browser's viewport and it can be used to implement this technique.
+
+[lazysizes](https://web.dev/use-lazysizes-to-lazyload-images/) is a popular JavaScript library for lazy-loading images and [`iframes`](http://afarkas.github.io/lazysizes/#examples). It supports YouTube embeds and [widgets](https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/unveilhooks). It also has [optional support](https://github.com/aFarkas/lazysizes/blob/097a9878817dd17be3366633e555f3929a7eaaf1/src/lazysizes-intersection.js) for IntersectionObserver. 
+
+{% Aside 'caution' %}
+
+Be careful when lazy-loading resources with JavaScript. If JavaScript fails to load, perhaps due to flaky network conditions, your resources won’t load at all. Keep an eye out for [native lazy-loading support for images and iframes](https://addyosmani.com/blog/lazy-loading/) that’s coming soon to Chrome.
+
+{% endAside %}
+
+## Optimize how you serve third-party scripts
+
+
+### Use third-party CDN hosting
+
+It’s common for third-party vendors to provide URLs of JavaScript files that they host, usually on a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network). The benefits of this approach are that you can get started quickly—just copy and paste the URL—and there’s no maintenance overhead. The third-party vendor handles server configuration and script updates.
+
+But because they are not on the same origin as the rest of your resources, loading files from a public CDN comes with a network cost. The browser needs to perform a DNS lookup, establish a new HTTP connection, and, on secure origins, perform an SSL handshake with the vendor’s server. 
+
+When you use files from third-party servers, you rarely have control over caching. Relying on someone else’s caching strategy might cause scripts to be unnecessarily re-fetched from the network too often.
+
+{% Aside %}
+
+One possible benefit of serving third-party code this way is cross-domain caching. Sometimes a resource served from a public CDN may already be cached in the user’s browser if they previously visited another site that used the same resource from the same public CDN. This optimization[ doesn’t happen in Safari](https://andydavies.me/blog/2018/09/06/safari-caching-and-3rd-party-resources/) due to security reasons and it’s hard to tell how common it is in other browsers.
```

Comment:
```suggestion
One possible benefit of using a CDN is cross-domain caching. Sometimes a resource served from a public CDN may already be cached in the user's browser if they previously visited another site that used the same resource. However, this optimization[ doesn't happen in Safari](https://andydavies.me/blog/2018/09/06/safari-caching-and-3rd-party-resources/), and it's hard to tell how common it is in other browsers.
```
@katejeffreys @mihajlija: I trimmed this a little. Aside from that, as noted above, I feel like we're kinda talking out of both sides of our mouths. I think maybe this needs to be moved to the body copy with another sentence basically saying "Look, odds are one of the two strategies below is a better bet." What do you think?